checker/src/org/checkerframework/checker/lock/LockAnnotatedTypeFactory.java

```diff
@@ -96,28 +107,310 @@ public LockTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, LockS
         return new LockTransfer((LockAnalysis) analysis,(LockChecker)this.checker);
     }
 
-    protected AnnotatedTypeMirror getDeclaredAndDefaultedAnnotatedType(Tree tree) {
-        shouldCache = false;
+    class LockQualifierHierarchy extends MultiGraphQualifierHierarchy {
+
+        public LockQualifierHierarchy(MultiGraphFactory f) {
+            super(f, LOCKHELD);
+        }
+
+
+        boolean isGuardedBy(AnnotationMirror am) {
+            return AnnotationUtils.areSameIgnoringValues(am, GUARDEDBY) ||
+                   AnnotationUtils.areSameIgnoringValues(am, JAVAXGUARDEDBY) ||
```

Comment:
In several places I see the code check the three different classes. Why is this? Why wasn't the earlier alias enough?
Can you abstract this behavior into a method and document why it is needed?